monai/transforms/intensity/array.py

```diff
@@ -857,20 +865,23 @@ def __init__(self, window_length: int, order: int, axis: int = 1, mode: str = "z
         self.axis = axis
         self.mode = mode
 
-    def __call__(self, img: np.ndarray):
+    def __call__(self, img: NdarrayTensor) -> NdarrayTensor:
         """
         Args:
-            img: numpy.ndarray containing input data. Must be real and in shape [channels, spatial1, spatial2, ...].
+            img: array containing input data. Must be real and in shape [channels, spatial1, spatial2, ...].
 
         Returns:
-            np.ndarray containing smoothed result.
+            array containing smoothed result.
 
         """
+        img_t: torch.Tensor
+        img_t, *_ = convert_data_type(img, torch.Tensor)  # type: ignore
+
         # add one to transform axis because a batch axis will be added at dimension 0
         savgol_filter = SavitzkyGolayFilter(self.window_length, self.order, self.axis + 1, self.mode)
```

Comment:
It would be slightly more efficient to store this as an instance member so it doesn't need to be recreated every time the transform is applied.